components/equihash/src/tromp.rs

```diff
@@ -30,23 +32,30 @@ extern "C" {
     fn equi_digiteven(eq: *mut CEqui, r: u32, id: u32);
     fn equi_digitK(eq: *mut CEqui, id: u32);
     fn equi_nsols(eq: *const CEqui) -> usize;
-    fn equi_sols(eq: *const CEqui) -> *const *const u32;
+    /// Returns `equi_nsols()` solutions of length `2^K`, in a single memory allocation.
+    fn equi_sols(eq: *const CEqui) -> *const u32;
```

Comment:
Despite its type in the C code, this is actually a single allocation containing multiple solutions. This fix avoids a segmentation fault.

https://github.com/zcash/librustzcash/blob/f3ea80e2abb1eef5cc38a554e7a60a90958262a4/components/equihash/tromp/equi_miner.c#L256

https://github.com/zcash/librustzcash/blob/f3ea80e2abb1eef5cc38a554e7a60a90958262a4/components/equihash/tromp/equi_miner.c#L208-L213